docker/lib/dependabot/docker/file_fetcher.rb

```diff
@@ -40,8 +40,9 @@ def fetch_files
           )
         elsif incorrectly_encoded_dockerfiles.none? && incorrectly_encoded_yamlfiles.none?
           raise(
-            Dependabot::DependabotError,
-            "Found neither Kubernetes YAML nor Dockerfiles in #{directory}"
+            Dependabot::DependencyFileNotFound,
+            File.join(directory, "Dockerfile"),
```

Comment:
I copied this from the implementation above where the feature flag is disabled... however they're both wrong because we look for any file matching the regex `DOCKER_REGEXP = /dockerfile/i`... 

I considered returning just the directory, but I noticed over in https://github.com/dependabot/dependabot-core/blob/b6784e245a6636982a5a66f764cf04cb0d3c2335/common/lib/dependabot/errors.rb#L86 that both `file_name` and `directory` methods assume a file name is tacked onto the path... 

I'm not sure where those methods are called, but I have a hunch they're displayed to the user in the error UI banner that we display... since we have custom handling of this error so that users don't have to grep through the logs to see why Dependabot is failing. So for now I just stuck with what exists and will investigate whether this is used at all later.